### PR TITLE
Fixes Jetpack Comments Validation Errors

### PR DESCRIPTION
**Pull request by shahthepro**
Fixes the issue reported here. https://wordpress.org/support/topic/please-fix-w3-validation-errors?replies=1&view=all

### CODE REVIEW

**Comment by jeherve** [comment]
It seems like your PR removes some of the changes introduced in 0d2908266b63b24d54693689fbcfa88e9de7d75d.

Could you make sure your PR only includes the changes you want to see in Jetpack?

Thanks!

**Comment by jeherve** [comment]
Related: #1071

**Comment by shahthepro** [comment]
@jeherve Check this commit https://github.com/shahthepro/jetpack/commit/4cbb498205a4e989b6f3d7a36f725fe8c035d7f4

It is what you meant, right?

**Comment by jeherve** [comment]
Yes, exactly! :+1:

File: modules/comments/comments.php [hunk 1]
```diff
@@ -272,7 +272,15 @@ public function comment_form_after() {
 		<div id="respond" class="comment-respond">
 			<h3 id="reply-title" class="comment-reply-title"><?php comment_form_title( esc_html( $params['greeting'] ), esc_html( $params['greeting_reply'] ) ); ?> <small><?php cancel_comment_reply_link( esc_html__( 'Cancel reply' , 'jetpack') ); ?></small></h3>
 			<div id="commentform" class="comment-form">
-				<iframe src="<?php echo esc_url( $url ); ?>" allowtransparency="<?php echo $transparent; ?>" style="width:100%; height: <?php echo $height; ?>px;border:0px;" frameBorder="0" scrolling="no" name="jetpack_remote_comment" id="jetpack_remote_comment"></iframe>
+				<iframe src="<?php echo esc_url( $url ); ?>" style="width:100%; height: <?php echo $height; ?>px; border:0; overflow:hidden;" name="jetpack_remote_comment" id="jetpack_remote_comment"></iframe>			
+				<!--[if !IE]><!-->
+				<script>
+				r(function(){
+					document.getElementById("jetpack_remote_comment").allowTransparency = <?php echo $transparent; ?>;
+				});
```
Comment (roccotripaldi) [inline modules/comments/comments.php]: I'm worried that there could be more than 1 comment form on a page, as specially if we are in a loop. In that case `getElementById` may not be reliable. Is there a CSS property that we can make use of?

Comment (shahthepro) [inline modules/comments/comments.php]: @roccotripaldi Check this commit https://github.com/Automattic/jetpack/commit/27926fe97701db5dfca0cca78a23eae1626559c0

Haven't tested it though. But, It should work however.

File: modules/comments/comments.php [hunk 2]
```diff
@@ -272,7 +272,19 @@ public function comment_form_after() {
 		<div id="respond" class="comment-respond">
 			<h3 id="reply-title" class="comment-reply-title"><?php comment_form_title( esc_html( $params['greeting'] ), esc_html( $params['greeting_reply'] ) ); ?> <small><?php cancel_comment_reply_link( esc_html__( 'Cancel reply' , 'jetpack') ); ?></small></h3>
 			<div id="commentform" class="comment-form">
-				<iframe src="<?php echo esc_url( $url ); ?>" allowtransparency="<?php echo $transparent; ?>" style="width:100%; height: <?php echo $height; ?>px;border:0px;" frameBorder="0" scrolling="no" name="jetpack_remote_comment" id="jetpack_remote_comment"></iframe>
+				<iframe src="<?php echo esc_url( $url ); ?>" style="width:100%; height: <?php echo $height; ?>px; border:0; overflow:hidden;" name="jetpack_remote_comment" class="jetpack_remote_comment" id="jetpack_remote_comment"></iframe>
```
Comment (eliorivero) [inline modules/comments/comments.php]: @shahthepro is the `overflow: hidden` necessary? it wasn't there previously and while testing this, I removed it and didn't notice any issue.

File: modules/comments/comments.php [hunk 3]
```diff
@@ -272,7 +272,19 @@ public function comment_form_after() {
 		<div id="respond" class="comment-respond">
 			<h3 id="reply-title" class="comment-reply-title"><?php comment_form_title( esc_html( $params['greeting'] ), esc_html( $params['greeting_reply'] ) ); ?> <small><?php cancel_comment_reply_link( esc_html__( 'Cancel reply' , 'jetpack') ); ?></small></h3>
 			<div id="commentform" class="comment-form">
-				<iframe src="<?php echo esc_url( $url ); ?>" allowtransparency="<?php echo $transparent; ?>" style="width:100%; height: <?php echo $height; ?>px;border:0px;" frameBorder="0" scrolling="no" name="jetpack_remote_comment" id="jetpack_remote_comment"></iframe>
+				<iframe src="<?php echo esc_url( $url ); ?>" style="width:100%; height: <?php echo $height; ?>px; border:0; overflow:hidden;" name="jetpack_remote_comment" class="jetpack_remote_comment" id="jetpack_remote_comment"></iframe>			
+				<!--[if !IE]><!-->
+				<script>
+				r(function(){
+					//document.getElementById("jetpack_remote_comment").allowTransparency = <?php echo $transparent; ?>;
```
Comment (eliorivero) [inline modules/comments/comments.php]: @shahthepro this comment can be removed.

File: modules/comments/comments.php [hunk 4]
```diff
@@ -272,7 +272,19 @@ public function comment_form_after() {
...
+					var comment_form_elements = document.getElementsByClassName("jetpack_remote_comment");
+					for (var i = 0; i < comment_form_elements.length; i++) {
+					    comment_form_elements[i].allowTransparency = <?php echo $transparent; ?>;
```
Comment (eliorivero) [inline modules/comments/comments.php]: @shahthepro after adding your change in a test site, the form flickers when it slides down as can be seen in this video https://cloudup.com/cizAOCNEbct
This is noticeable whether user is logged in or out.
After this line that adds the transparency, I added
```js
comment_form_elements[i].scrolling = 'no';
```
and it solved the flickering.

Also, can you format this following [JavaScript Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/)? Please add spacing, use single quotes for string, use meaningful names.

Finally, this JavaScript block declares `r()` with global scope and could conflict with something so please wrap this in a closure.

Thanks!

**Comment by samhotchkiss** [comment]
Moved out of the pipeline until @shahthepro addresses @eliorivero's concerns and rebases

**Comment by samhotchkiss** [comment]
@eliorivero -- if you want to go ahead and make the changes you recommended here, that'd be great!

**Comment by eliorivero** [comment]
Closing in favor of https://github.com/Automattic/jetpack/pull/6316 where the commits from author have been cherry picked.

**Pull request closed without merging**